src/gp/src/GPMSA.C

```diff
@@ -1073,7 +1071,30 @@ GPMSAFactory<V, M>::setUpEmulator()
               unsigned int j = ex +
                 m_numExperiments * (num_discrepancy_bases + outj);
 
-              B(i,j) = (*m_TruncatedSVD_simulationOutputs)(outi,outj);
+              V & singular_vector = m_TruncatedSVD_simulationOutputs[outj];
+              // If there's just multivariate data here, then no
+              // interpolation is necessary.
+              double Kvalue = singular_vector[outi];
+
+              if (this->m_experimentVariables.size())
+                {
+                  const unsigned int var = this->m_experimentVariables[ex][outi];
+                  const SimulationOutputPoint & pt = this->m_experimentPoints[ex][outi];
+
+                  if (var < this->m_simulationMeshes.size())
+                    {
+                      SimulationOutputMesh<V> & mesh =
+                        *this->m_simulationMeshes[var];
+                      Kvalue =
+
+                        mesh.interpolateOutput(singular_vector, pt);
+                    }
+                  else
+                    Kvalue =
+                      singular_vector[var + variable_index_to_output_index];
+                }
+
+              B(i,j) = Kvalue;
```

Comment:
And this doesn't regress anything?  That's awesome.

Reply:
None of my commits were made before tests were passed.  No idea about the intermediate stages after rebase-after-rebase, though.

Reply:
Yeah that's fine.